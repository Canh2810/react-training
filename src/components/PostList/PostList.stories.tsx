import { StoryObj, Meta } from '@storybook/react'
import PostList from '.'
import { POSTS_MOCK } from '@constants'
import { MemoryRouter } from 'react-router-dom'

export default {
  title: 'Components/PostList',
  component: PostList,
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
} as Meta

type Story = StoryObj<typeof PostList>

export const Default: Story = {
  args: {
    posts: POSTS_MOCK,
  },
}
