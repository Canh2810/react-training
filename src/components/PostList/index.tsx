// Libs
import { memo, useCallback } from 'react'
import { useNavigate } from 'react-router-dom'

// Components
import { PostCard } from '../index'

// Types
import { IPost } from '@types'

// Style
import './index.css'
import { ROUTES } from '@constants'
import { useQueryStore } from '@stores'

export interface PostListProps {
  posts: IPost[]
}

const PostList = ({ posts }: PostListProps) => {
  const navigate = useNavigate()
  const { clearQuery } = useQueryStore((state) => state)

  const handleClickPostCard = useCallback(
    (id: string) => {
      navigate(ROUTES.SINGLE_POST_PARAM + id)
      clearQuery()
    },
    [clearQuery],
  )

  return (
    <div className="post-list">
      {posts.map((post: IPost) => {
        const { id } = post || {}
        const handleClick = () => handleClickPostCard(id)
        return <PostCard key={id} post={post} onClick={handleClick} />
      })}
    </div>
  )
}

export default memo(PostList)
