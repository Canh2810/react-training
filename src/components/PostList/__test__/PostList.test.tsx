// PostList.test.js (or PostList.test.tsx for TypeScript)

import { fireEvent, render } from '@testing-library/react'
import PostList from '..'
import { POSTS_MOCK, ROUTES } from '@constants'
import { MemoryRouter } from 'react-router-dom'

// Mock the useNavigate hook
const mockNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}))

describe('PostList component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <PostList posts={POSTS_MOCK} />
      </MemoryRouter>,
    )
    expect(container).toMatchSnapshot()
  })

  it('navigates to a single post when a post card is clicked', () => {
    const { getByText } = render(
      <MemoryRouter>
        <PostList posts={POSTS_MOCK} />
      </MemoryRouter>,
    )
    fireEvent.click(getByText(POSTS_MOCK[1].title))
    expect(mockNavigate).toHaveBeenCalledWith(
      ROUTES.SINGLE_POST_PARAM + POSTS_MOCK[1].id,
    )
  })
})
