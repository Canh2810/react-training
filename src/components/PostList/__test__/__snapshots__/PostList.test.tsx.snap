// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList component should render correctly 1`] = `
<div>
  <div
    class="post-list"
  >
    <div
      class="post-card"
    >
      <img
        alt="The Impact of Technology on the Workplace: How Technology is Changing"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          The Impact of Technology on the Workplace: How Technology is Changing
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://www.w3schools.com/w3images/avatar2.png"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="Beta Testing: The final decisive step before launching the software"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1539667468225-eebb663053e6?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MjF8fG5hdHVyYWx8ZW58MHx8MHx8fDA%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          Beta Testing: The final decisive step before launching the software
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Canh Vo"
            class="post-card__author-avatar"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSEJuOqDYeLdcJOq8OctOVOCWp-RTFkdmWK4dcI5F9YLg&s"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Canh Vo
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="Make a placeholder content in SwiftUI with Redacted & Unredacted"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          Make a placeholder content in SwiftUI with Redacted & Unredacted
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSHeu1JU1avFzQtN-az4HZqEpR1VvEBN-SEXfEuOtt8Xg&s"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="Instructions for Security Testing using OWASP ZAP tool for penetration testers - Penetration testing"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          Instructions for Security Testing using OWASP ZAP tool for penetration testers - Penetration testing
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://www.w3schools.com/w3images/avatar2.png"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="Excel at Cashiering with the McDonald POS Training Simulator APK!"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          Excel at Cashiering with the McDonald POS Training Simulator APK!
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://www.w3schools.com/w3images/avatar2.png"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="The Impact of Technology on the Workplace: How Technology is Changing"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1418489098061-ce87b5dc3aee?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTh8fG5hdHVyYWx8ZW58MHx8MHx8fDA%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          The Impact of Technology on the Workplace: How Technology is Changing
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQIgYqjNVQSYlcOONsKMfyvLJWY-MGgtqwfqMx7-2kjzg&s"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="Excel at Cashiering with the McDonald POS Training Simulator APK!"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1539667468225-eebb663053e6?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MjF8fG5hdHVyYWx8ZW58MHx8MHx8fDA%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          Excel at Cashiering with the McDonald POS Training Simulator APK!
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRlHJmFah1lW88FU0D-fD9m-0HXQGHd5QfD8GiL4guU-g&s"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="The Impact of Technology on the Workplace: How Technology is Changing"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1418489098061-ce87b5dc3aee?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTh8fG5hdHVyYWx8ZW58MHx8MHx8fDA%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          The Impact of Technology on the Workplace: How Technology is Changing
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://www.w3schools.com/w3images/avatar2.png"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="The Impact of Technology on the Workplace: How Technology is Changing"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          The Impact of Technology on the Workplace: How Technology is Changing
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://www.w3schools.com/w3images/avatar2.png"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="The Impact of Technology on the Workplace: How Technology is Changing"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          The Impact of Technology on the Workplace: How Technology is Changing
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://www.w3schools.com/w3images/avatar2.png"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="The Impact of Technology on the Workplace: How Technology is Changing"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          The Impact of Technology on the Workplace: How Technology is Changing
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://www.w3schools.com/w3images/avatar2.png"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="The Impact of Technology on the Workplace: How Technology is Changing"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          The Impact of Technology on the Workplace: How Technology is Changing
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://www.w3schools.com/w3images/avatar2.png"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="The Impact of Technology on the Workplace: How Technology is Changing"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1539667468225-eebb663053e6?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MjF8fG5hdHVyYWx8ZW58MHx8MHx8fDA%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          The Impact of Technology on the Workplace: How Technology is Changing
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSHeu1JU1avFzQtN-az4HZqEpR1VvEBN-SEXfEuOtt8Xg&s"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
    <div
      class="post-card"
    >
      <img
        alt="Excel at Cashiering with the McDonald POS Training Simulator APK!"
        class="post-card__img"
        src="https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60"
      />
      <div
        class="post-card__content"
      >
        <span
          class="badge"
          style="background-color: rgba(75, 107, 251, 0.05); color: rgb(75, 107, 251);"
        >
          Technology
        </span>
        <p
          class=""
          style="font-family: work-sans; font-size: 24px; font-weight: 600; line-height: 28px; color: rgb(59, 60, 74);"
        >
          Excel at Cashiering with the McDonald POS Training Simulator APK!
        </p>
        <div
          class="post-card__footer"
        >
          <img
            alt="Tracey Wilson"
            class="post-card__author-avatar"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSEJuOqDYeLdcJOq8OctOVOCWp-RTFkdmWK4dcI5F9YLg&s"
          />
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            Tracey Wilson
          </p>
          <p
            class=""
            style="font-family: work-sans; font-size: 18px; font-weight: 400; line-height: 26px; color: rgb(59, 60, 74);"
          >
            August 20, 2022
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
